### PR TITLE
Exception list React implementation

### PR DESCRIPTION
**Pull request by jmartinez-tacc**
## Overview

Exception list React implementation

## Related

[WP-635](https://tacc-main.atlassian.net/browse/WP-635)

## Changes

Exception list and modals conversion.

## Testing

1. Navigate to http://localhost:8000/administration/list-exceptions/
2. Select filters and make sure those filters are applied to the Exceptions list
3. From Actions drop-down select View Record, make sure the information is displayed
4. From Actions drop-down select Edit Record, make sure the information is updated and changes are reflected on View Record modal window and listing

## UI

Screenshot 2024-10-15 at 2 52 25 PM
Screenshot 2024-10-15 at 2 52 42 PM
Screenshot 2024-10-15 at 2 52 58 PM
Screenshot 2024-10-15 at 2 53 08 PM

### CODE REVIEW

**Review by chandra-tacc — CHANGES_REQUESTED** [review]
The edit modal for exception has differences between localhost and [dev.apcd.](https://dev.apcd.tacc.utexas.edu/administration/list-exceptions/)

There is header and field list is different in dev and production. See below
Screenshot 2024-10-16 at 10 27 53 AM

File: apcd-cms/src/client/src/components/Admin/EditExceptionModal/EditExceptionModal.module.css
Comment (sophia-massie) [inline apcd-cms/src/client/src/components/Admin/EditExceptionModal/EditExceptionModal.module.css:1]: Here are some styles currently being applied to the modal and listing part of the page before your style module (EditExceptionModal.module.css) https://github.com/TACC/Core-CMS-Custom/tree/main/apcd_cms/src/apps/admin_exception/static/admin_exception/css

File: apcd-cms/src/client/src/components/Admin/ViewExceptionModal/ViewExceptionModal.module.css
Comment (sophia-massie) [inline apcd-cms/src/client/src/components/Admin/ViewExceptionModal/ViewExceptionModal.module.css:1]: Here are some styles currently being applied to the modal that could be conflicting with is. https://github.com/TACC/Core-CMS-Custom/tree/main/apcd_cms/src/apps/admin_exception/static/admin_exception/css

File: apcd-cms/src/apps/admin_exception/templates/view_admin_exception_modal.html [hunk 1]
```diff
@@ -1,65 +1,11 @@
+{% extends "apcd_cms/templates/standard.html" %}
 {% load static %}
 <link rel="stylesheet" href="{% static 'apcd-cms/css/modal.css' %}">
```
Comment (sophia-massie) [inline apcd-cms/src/apps/admin_exception/templates/view_admin_exception_modal.html:3]: Some of your style modules (ViewExceptionModal.module.css in this case) may be clashing with these stylesheets here and might be why the modals don't look the same as dev/staging/prod.

**Review by sophia-massie — CHANGES_REQUESTED** [review]
LGTM here are some ideas for styling. May be a good idea as a follow on to eventually factor out global element styles (modal, input, table, etc) so that all table listing, form fields, and modals start with the same basic layout and then have that sheet applied to main.tsx

**Review by chandra-tacc — APPROVED** [review]
approved in current state with known follow up bugs

**Pull request merged**